API.md

```diff
@@ -0,0 +1,141 @@
+
+```shell
+$ oc get flightrecorders
+NAME           AGE
+containerjfr   7d16h
+```
+
+`FlightRecorder` objects are created by the operator whenever a new Container JFR-compatible service is detected.
+Services that expose a port named `jfr-jmx` are considered compatible. The number of this port is stored in the `spec.port` property for use by the operator. Each `FlightRecorder` object maps one-to-one with a Kubernetes service. This service is stored in the `status.target` property of the `FlightRecorder` object.
+
+```shell
+$ oc get -o json flightrecorders/containerjfr
+```
+```json
+{
+    "apiVersion": "rhjmc.redhat.com/v1alpha1",
+    "kind": "FlightRecorder",
+    "metadata": {
+        "creationTimestamp": "2020-01-15T22:54:22Z",
+        "generation": 1,
+        "labels": {
+            "app": "containerjfr"
+        },
+        "name": "containerjfr",
+        "namespace": "default",
+        "ownerReferences": [
+            {
+                "apiVersion": "v1",
+                "blockOwnerDeletion": true,
+                "controller": true,
+                "kind": "Service",
+                "name": "containerjfr",
+                "uid": "f895e203-37e9-11ea-8866-52fdfc072182"
+            }
+        ],
+        "resourceVersion": "298903",
+        "selfLink": "/apis/rhjmc.redhat.com/v1alpha1/namespaces/default/flightrecorders/containerjfr",
+        "uid": "f896b6e1-37e9-11ea-8866-52fdfc072182"
+    },
+    "spec": {
+        "port": 9091,
+        "recordingRequests": []
+    },
+    "status": {
+        "recordings": [],
+        "target": {
+            "apiVersion": "v1",
+            "kind": "Service",
+            "name": "containerjfr",
+            "namespace": "default",
+            "resourceVersion": "298899",
+            "uid": "f895e203-37e9-11ea-8866-52fdfc072182"
+        }
+    }
+}
+```
+
+## Creating a new Flight Recording
+
+To start a new recording, you will need to add a new recording request to the `FlightRecorder` in the `spec.recordingRequests` array. You can do this nicely on the command line with `oc edit`. The recording request must include the following:
+
+1. `name`: a string uniquely identifying the recording within that service
+2. `eventOptions`: an array of string options passed to Container JFR. The `"ALL"` special string can be used to enable all available events.
+3. `duration`: length of the requested recording as a [duration string](https://golang.org/pkg/time/#ParseDuration).
+
+```shell
+oc edit -o json flightrecorders/containerjfr
+```
+```json
+    "spec": {
+        "port": 9091,
+        "recordingRequests": [
+                {
+                        "name": "my-recording",
+                        "eventOptions": [ "jdk.SocketRead:enabled=true", "jdk.SocketWrite:enabled=true" ],
+                        "duration": "30s"
+                }
+        ]
+    },
+    "status": {
+        "recordings": [],
```

Comment:
Looks like the JSON lost some braces and such here

Reply:
This was actually cropped to only show the relevant parts of the JSON. I'll add the missing closing braces at least.

Reply:
This should be fixed now.